lib/src/http/browser/request_mixin.dart

```diff
@@ -25,49 +25,50 @@ import 'package:w_transport/src/http/request_exception.dart';
 import 'package:w_transport/src/http/response.dart';
 
 abstract class BrowserRequestMixin implements BaseRequest, CommonRequest {
-  HttpRequest _request = HttpRequest();
+  HttpRequest? _request;
 
   @override
   void abortRequest() {
-    _request.abort();
+    _request?.abort();
   }
 
   @override
   Future<Null> openRequest([_]) async {
     _request = HttpRequest();
-    _request.open(method!, uri.toString());
+    _request!.open(method!, uri.toString());
   }
 
   @override
   Future<BaseResponse> sendRequestAndFetchResponse(
       FinalizedRequest finalizedRequest,
       {bool streamResponse = false}) async {
     final c = Completer<BaseResponse>();
+    final request = _request!;
```

Comment:
Since it's final, it shouldn't be a problem. It's just a place to store the same reference as `_request` so that we get the type promotion to non-nullable, but when `abortRequest` is called it will operate on the same request instance.